compiler_opt/rl/corpus.py

```diff
@@ -343,10 +343,20 @@ def __init__(self,
 
     # don't use add/remove for replace
     add_keys = {k.split('=', maxsplit=1)[0] for k in additional_flags}
-    if add_keys.intersection(
-        set(replace_flags)) or set(delete_flags).intersection(
-            set(replace_flags)) or add_keys.intersection(set(delete_flags)):
-      raise ValueError('do not use add/delete flags to replace')
+    if add_keys.intersection(set(replace_flags)):
+      raise ValueError(
+          'Do not use add/delete flags to replace flags: add flags and '
+          'replace flags intersect: '
+          f'{add_keys.intersection(set(replace_flags))}')
+    elif set(delete_flags).intersection(set(replace_flags)):
+      raise ValueError(
+          'Do not use add/delete flags to replace flags: delete flags and '
```

Comment:
is the string the exact same? factor it out, e.g

msg = 'Do not use add/delete flags to replace flags: add flags and '
          'replace flags intersect: '
          f'{details}'

and then you can `msg.format(details = <whatever it is.. .add_keys.intersection... whatever>)`

Reply:
They're not exactly the same (what intersects differs), and I think factoring it out would just make the code quite a bit less readable. Happy to follow up post commit if opinions continue to differ though.